daemon/routes/iprule/iprule.go

```diff
@@ -368,22 +368,30 @@ func checkFwmarkRule(fwMarkVal uint32, ipv6 bool) (bool, error) {
 	// #nosec G204 -- input is properly sanitized
 	out, err := exec.Command(cmdStr, cmdParams...).CombinedOutput()
 	if err != nil {
-		return false, fmt.Errorf("executing '%s %s' command: %w: %s", cmdStr, strings.Join(cmdParams, " "), err, string(out))
+		return false, 0, fmt.Errorf("executing '%s %s' command: %w: %s", cmdStr, strings.Join(cmdParams, " "), err, string(out))
 	}
 
-	lookupStr := fmt.Sprintf("not from all fwmark 0x%x", fwMarkVal)
+	lookupStr := fmt.Sprintf("not from all fwmark 0x%x lookup", fwMarkVal)
 
 	// parse ip cmd output line-by-line
 	for _, line := range bytes.Split(out, []byte{'\n'}) {
 		if len(line) == 0 {
 			continue
 		}
 		if strings.Contains(string(line), lookupStr) {
-			return true, nil
+			words := strings.Split(strings.Trim(string(line), " "), " ")
+			if len(words) < 7 || words[len(words)-2] != "lookup" {
+				return false, 0, fmt.Errorf("failed to find fwmark rule: '%s'", line)
+			}
+			tbldID, err := strconv.ParseUint(words[len(words)-1], 10, 64)
+			if err != nil {
+				return false, 0, fmt.Errorf("converting '%s' to uint: %w", words[len(words)-1], err)
+			}
+			return true, uint(tbldID), nil
 		}
 	}
 
-	return false, nil
+	return false, 0, nil
```

Comment:
It might be better to return some custom created error in this case, e.g. `not found`, and the caller to check the error type and decide what to do?